### DISCUSSION TITLE
Similarity with Asphalt

### DISCUSSION THREAD

**Discussion by davidbrochart** [hynek/svcs#76] [Q&A]
Great project!
Do you know [Asphalt](https://github.com/asphalt-framework/asphalt)? Both projects seem to solve the same kind of problems, although Asphalt is not just for Web frameworks.
Maybe some inspiration to take from each other?

**Comment by davidbrochart** [comment]
Reading more about svcs, it doesn't seem to be only for Web frameworks either. But what I can see that is missing in svcs is configuration? See Asphalt's [documentation](https://asphalt.readthedocs.io/en/latest/userguide/deployment.html).

**Reply by hynek** [reply]
Yeah, *svcs* is decidedly not web-only, it's just the easiest to talk about. AFAIK, Asphalt is a fully-fledged application framework while *svcs* is a pluggable service location framework.

Does it have a service location part and if so – does it something interesting *svcs* doesn't?

**Reply by davidbrochart** [reply]
I think service location in Asphalt is provided through [contexts](https://asphalt.readthedocs.io/en/latest/userguide/contexts.html). In [Asphalt v5.0](https://github.com/asphalt-framework/asphalt/tree/5.0) there is a lot of work to manage async tasks and have them play well with AnyIO (from the same author). But I don't know if it provides something *svcs* doesn't.
Do you think configuration could be added to *svcs*, or is it not possible because it is not an application framework?

**Reply by hynek** [reply]
Yeah, contexts sound like *svcs* will become once #70 is implemented, but you’ll notice a very different API/philosophy. For example how it handles cleanups, or how it leverages injection for type-safety.

Overall, I would describe it that both implement the same ideas (well, I still have to implement #70 😅) but follow different design philosophies.

> Do you think configuration could be added to svcs, or is it not possible because it is not an application framework?

I’m not sure what you mean by that. *svcs* is designed for integration, but Asphalt already has something similar?

**Reply by davidbrochart** [reply]
Great to see that both projects converge on some ideas (with #70), even though there are differences.
Yes Asphalt has a configuration system, but I think it would have to be done on top of *svcs*, since *svcs* is designed for integration and is not an application framework per say.
I really like the efforts that went into the definition of concepts in *svcs*'s documentation, it is very helpful. Also the [video](https://www.youtube.com/watch?v=d1elMD9WgpA) was very nice :+1:
